Marksman/Toc.fs

```diff
@@ -50,8 +54,24 @@ module TableOfContents =
         else
             Some { entries = Array.map Entry.fromHeading headings }
 
+    let insertionPoint (doc: Workspace.Doc) : InsertionPoint =
+        match (Array.toList doc.index.titles) with
+        // if there's only a single title
+        | [ singleTitle ] -> After singleTitle.range
+        | _ ->
+            match doc.index.yamlFrontMatter with
+            | None -> Replacing Text.documentBeginning
+            | Some (yml) -> After yml.range
```

Comment:
This is the main bit of logic to bikeshed about

Reply:
Not much to bike shed. I think this makes total sense.